plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -1427,6 +1427,10 @@ void producer_plugin::plugin_startup() {
 void producer_plugin_impl::plugin_shutdown() {
    boost::system::error_code ec;
    _timer.cancel(ec);
+   boost::system::error_code ro_ec;
```

Comment:
`ec` is a out param, but doesn't feel right to reuse it to me. Since we ignore it, it doesn't matter either way. I fine either way. Want me to change it?

Reply:
I feel that when it doesn't matter either way, we should opt for simplicity & shorter code, but I'll leave it up to you as it is a matter of personal preference.

Reply:
Done.